apis/pipelines/hub/test_generators.go

```diff
@@ -63,25 +63,32 @@ func RandomPipelineSpec(provider common.NamespacedName) PipelineSpec {
 }
 
 func RandomProvider() *Provider {
+
 	return &Provider{
 		TypeMeta: metav1.TypeMeta{},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      RandomLowercaseString(),
 			Namespace: "default",
 		},
 		Spec:   RandomProviderSpec(),
-		Status: RandomStatus(common.NamespacedName{}),
+		Status: RemoveProvider(RandomStatus(common.NamespacedName{})),
 	}
 }
 
+// RemoveProvider removes the provider from the status
+// This is because providers never have this field populated, nor should they.
```

Comment:
maybe just clarify that the status type is used by other resources hence why you can't just remove it from `RandomStatus`?
```suggestion
// This is because providers never have this field populated, nor should they, but share the Status type with other resources that do use this field.
```

Reply:
Shouldn't be sharing types when all the fields aren't being used in every type it's being shared with to begin with tbh

Reply:
have added in latest changes